File: src/main/resources/public/js/factories/SocketMessageFactory.js
/**
 * Created by manasb on 15-11-2016.
 */

let MessageType = require("../domain/MessageType");
let ChatMessage = require ("../domain/ChatMessage");

class SocketMessageFactory {

    static createChatMessage(user, message) {
        let chatMessage = new ChatMessage(user, message, Date.now());
        return JSON.stringify({
            messageType: MessageType.CHAT_MESSAGE,
            data: chatMessage
        });
    }
}

module.exports = SocketMessageFactory;

File: server/src/main/java/domain/SocketMessageFactory.java
package domain;

import java.util.List;

/**
 * Created by manasb on 26-12-2016.
 */
public final class SocketMessageFactory {

    public static SocketMessage<ChatMessage> createChatMessage(final ChatMessage chatMessage) {
        return new SocketMessage<>(MessageType.CHAT_MESSAGE, chatMessage);
    }

    public static SocketMessage<String> createErrorMessage(final String error) {
        return new SocketMessage<>(MessageType.ERROR, error);
    }

    public static SocketMessage<List<User>> createConnectedUsersMessage(final List<User> connectedUsers) {
        return new SocketMessage<>(MessageType.CONNECTED_USERS, connectedUsers);
    }
}
